lib/msf/core/post/file.rb

```diff
@@ -307,38 +329,54 @@ def upload_file(remote, local)
   #
   # Delete remote files
   #
+  # @param remote_files [Array<String>] List of remote filenames to
+  #   delete
+  # @return [void]
   def rm_f(*remote_files)
     remote_files.each do |remote|
       if session.type == "meterpreter"
         session.fs.file.delete(remote) if exist?(remote)
       else
         if session.platform =~ /win/
-          cmd_exec("del /q /f #{remote}")
+          cmd_exec("del /q /f \"#{remote}\"")
         else
-          cmd_exec("rm -f #{remote}")
+          cmd_exec("rm -f '#{remote}'")
```

Comment:
Path quoting is still pretty inconsistent - sometimes it's ", other times it's '. Making it all \" should be the standard so shell expansion can work, e.g. rm -f ${USER}-tmp.